src/api/dynamic_data.c

```diff
@@ -302,28 +370,24 @@ rosidl_dynamic_typesupport_dynamic_data_destroy(
 rcutils_ret_t
 rosidl_dynamic_typesupport_dynamic_data_serialize(
   rosidl_dynamic_typesupport_dynamic_data_t * dynamic_data,
-  rcutils_uint8_array_t * buffer,
-  bool * success)
+  rcutils_uint8_array_t * buffer)
 {
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(dynamic_data, RCUTILS_RET_INVALID_ARGUMENT);
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(buffer, RCUTILS_RET_INVALID_ARGUMENT);
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(success, RCUTILS_RET_INVALID_ARGUMENT);
-  return (dynamic_data->serialization_support->methods->dynamic_data_serialize)(
-    dynamic_data->serialization_support->impl, dynamic_data->impl, buffer, success);
+  return (dynamic_data->serialization_support->methods.dynamic_data_serialize)(
+    &dynamic_data->serialization_support->impl, &dynamic_data->impl, buffer);
```

Comment:
Don't we also need a pull request in https://github.com/ros2/rosidl_dynamic_typesupport_fastrtps to update `fastrtps__dynamic_data_serialize` as well?

(same goes for deserialize)

Reply:
You're right, I forgot to spin up the PR.. I thought I caught them all :man_facepalming: 
Sorry about that

https://github.com/ros2/rosidl_dynamic_typesupport_fastrtps/pull/3